ethereumj-core/src/main/java/org/ethereum/facade/Ethereum.java

```diff
@@ -53,8 +56,27 @@
 
     Blockchain getBlockchain();
 
+    /**
+     * Uses expensive enough logic for delivery events to client.<br>
+     * Will be removed in future releases.
+     *
+     * @param listener
+     * @deprecated use {@link #subscribe(Subscription)} instead.
+     */
+    @Deprecated
     void addListener(EthereumListener listener);
 
+    /**
+     * Adds subscription for specific event.
+     *
+     * @param subscription
+     * @return publisher instance to support fluent API.
+     * @see Subscription
+     * @see Publisher
+     * @see Event
+     */
+    Publisher subscribe(Subscription subscription);
```

Comment:
As a user I'd rather expect `subscribe(Class type, Consumer handler);` which is clearer. But we have to use generics though to make it really convenient:
`<T> Publisher subscribe(Class<? extends Event<T>> type, Consumer<T> handler);`

Reply:
I'll add two overloaded methods: 
 * `<T> Publisher subscribe(Class<? extends Event<T>> type, Consumer<T> handler);`
 * `<T> Publisher subscribe(Class<? extends Event<T>> type, BiConsumer<T, Subscription.LifeCycle> handler);
`
but we need current method too, cause sometimes we are in want of `Subscription` instance.

Reply:
We should add some handy links where user could see available type of Events. Current user entrance could stuck without samples.

Reply:
Each of samples migrated to the new feature, so I guess it's not problem. Also we have utility class `Events` for convenience instantiation all of supported events. I'd rather actualize `Publisher` javadoc to refer user to all necessary classes.

Reply:
Before this update user was adding listener by extending `EthereumListenerAdapter`, he got the view of all available listening options in one click. Since this update he should write `.subscribe(` and what's next? Samples are not the answer, not everyone can find them and you will not go to samples every time you need to add listener, Subscription class looks difficult from the first view, events are different classes in some package somewhere, not listed in one place. So the idea is that main entrance should look handy and simplify user's usage. I'm not sure what's the best way to do it but current entrance could stuck most of users.